conans/client/recorder/action_recorder.py

```diff
@@ -41,15 +41,18 @@ def __init__(self):
 
     # ###### INSTALL METHODS ############
     def add_recipe_being_developed(self, reference):
+        assert(isinstance(reference, ConanFileReference))
         self._inst_recipes_develop.add(reference)
 
     def _add_recipe_action(self, reference, action):
         assert(isinstance(reference, ConanFileReference))
+        reference = reference.copy_clear_rev()
```

Comment:
I think that we should keep the revision and maybe improve the `not in` search below. But having the revision here might be useful in the future.

Reply:
I tried to change it but it is not obvious. If you receive an action for a reference (without revision) and then another one with revision, what the `key` should be... giving that in local the "revision" is not a thing except the metadata, maybe in the future we can add a field somewhere with the revision, but I would keep the keys as references without revision.